src/integration/java/io/pinecone/AfterAllTestsListener.java

```diff
@@ -0,0 +1,22 @@
+package io.pinecone;
+
+import io.pinecone.helpers.IndexManagerSingleton;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestPlan;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AfterAllTestsListener implements TestExecutionListener {
+    private static final Logger logger = LoggerFactory.getLogger(AfterAllTestsListener.class);
+
+    @Override
+    public void testPlanExecutionStarted(TestPlan testPlan) {
+        logger.info("Integration tests starting. Running setup...");
+
```

Comment:
There is nothing to do on startup? Is that because resources are created on-demand throughout the test run?

Reply:
Yeah, since it's a singleton and we're lazily creating index resources as needed, we only need to call for cleanup at the end of our test run at the moment.